lib/inspec/profile_context.rb

```diff
@@ -4,17 +4,20 @@
 require 'inspec/log'
 require 'inspec/rule'
 require 'inspec/dsl'
+require 'inspec/resource'
 require 'inspec/require_loader'
 require 'securerandom'
 require 'inspec/objects/attribute'
 
 module Inspec
   class ProfileContext # rubocop:disable Metrics/ClassLength
     def self.for_profile(profile, backend)
-      new(profile.name, backend, { 'profile' => profile })
+      c = new(profile.name, backend, { 'profile' => profile })
+      c.load_profile_content(profile)
     end
 
-    attr_reader :attributes, :rules, :profile_id
+    attr_reader :attributes, :rules, :profile_id, :resource_registry
+    attr_reader :profile_execution_context
```

Comment:
Based on our discussion I think we'll end up with two context, perhaps:

- `control_eval_context`
- `resource_eval_context`